src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -120,18 +120,18 @@ public void save(
             caseRepository.saveCaseEvent(authorization, updatedClaim, SETTLED_PRE_JUDGMENT);
         }
 
-        Optional<CaseEvent> caseEvent = Optional.empty();
+        CaseEvent caseEvent = null;
         if (claimantResponse.getType() == REJECTION) {
             caseEvent = directionsQuestionnaireService.prepareCaseEvent(
                 (ResponseRejection) claimantResponse,
                 updatedClaim
             );
-            if (caseEvent.isPresent()) {
-                caseRepository.saveCaseEvent(authorization, updatedClaim, caseEvent.get());
+            if (caseEvent != null) {
```

Comment:
We can move the `saveCaseEvent` call into the `if (claimantResponse.getType() == REJECTION) {` statement and remove the null check